src/freertos_drivers/st/Stm32Can.cxx

```diff
@@ -103,13 +132,58 @@ Stm32Can::Stm32Can(const char *name)
     : Can(name)
     , state_(CAN_STATE_STOPPED)
 {
-    /* only one instance allowed */
-    HASSERT(instances[0] == NULL);
+    /* Get dev num (digit at the end of the devname). */
```

Comment:
I don't like that we have programmatically tied the device "name" to the hardware instance. As far as I recall, we don't have any such tie anywhere else in the device file system. I think it makes since to instead pass the hardware references in (base address and interrupt number), similar to what we do in the Tiva driver: https://github.com/bakerstu/openmrn/blob/e35c58d4aba4b9d5a4dc355684a2eb1c87008f01/src/freertos_drivers/ti/TivaCan.cxx#L58

I strongly encourage that with the addition of these parameters, either we have an overloaded constructor that is backwards compatible fixed to the first CAN instance, or we use "default" values for the parameters to point to the first CAN instance. That way we don't break other existing uses.

Reply:
Done. The constructor takes an option device index that defaults to 0 (first/only CAN IF).